src/users.rs

```diff
@@ -273,6 +305,18 @@ impl OperateUsers for Auth0Client {
         .await?;
         Ok(())
     }
+
+    async fn check_password(&mut self, payload: &CheckPasswordPayload) -> Auth0Result<()> {
+        self.grant_type(GrantType::Password);
+
+        self.authenticate(Some(Credentials {
+            username: payload.username.clone().ok_or(Error::InvalidUsername)?,
+            password: payload.password.clone().ok_or(Error::InvalidPassword)?,
+        }))
```

Comment:
Using authenticate will erase previously stored credentials. We probably need another function or a condition in it.

Reply:
Fixed